packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -248,38 +249,69 @@ export function getOperationFunction(
   // Extract required parameters
   const parameters: OptionalKind<ParameterDeclarationStructure>[] =
     getOperationSignatureParameters(operation, clientType);
+  const isPaging = isPagingOperation(operation);
 
   // TODO: Support operation overloads
   const response = operation.responses[0]!;
-  const returnType = response?.type?.type
-    ? buildType(response.type.name, response.type, response.type.format)
-    : { name: "", type: "void" };
-
+  let returnType = { name: "", type: "void" };
+  if (response.type?.type) {
+    let type = response.type;
+    if (isPaging) {
+      type = extractPagingType(type, operation.itemName) ?? type;
+    }
+    returnType = buildType(type.name, type, type.format);
```

Comment:
I may not get your point here, currently if there is no `@items` defined we would not generate paging for them. Because we don't know the detailed item type.